x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -161,7 +163,8 @@ public boolean equals(Object obj) {
 
 
     private static final List<String> FIELD_NAMES_BLACKLIST = Arrays.asList("_size");
-    private static final String UNMAPPED = "unmapped";
+    private static final Set<String> COMPATIBLE_KEYWORD_TYPES = Set.of(KEYWORD.typeName(), CONSTANT_KEYWORD.typeName());
```

Comment:
Use a different constructor to make backporting easy  and potentially add the constant after existing content not in the middle of it.